azurerm/internal/services/compute/virtual_machine.go

```diff
@@ -283,6 +283,98 @@ func virtualMachineOSDiskSchema() *schema.Schema {
 	}
 }
 
+func virtualMachineDataDiskSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"caching": {
+					Type:     schema.TypeString,
+					Required: true,
+					ValidateFunc: validation.StringInSlice([]string{
+						string(compute.CachingTypesNone),
+						string(compute.CachingTypesReadOnly),
+						string(compute.CachingTypesReadWrite),
+					}, false),
+				},
+
+				"lun": {
+					Type:         schema.TypeInt,
+					Required:     true,
+					ValidateFunc: validation.IntAtLeast(0),
+				},
+
+				"name": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"storage_account_type": {
+					Type:     schema.TypeString,
+					Required: true,
+					ValidateFunc: validation.StringInSlice([]string{
+						string(compute.StandardLRS),
+						string(compute.PremiumLRS),
+						string(compute.StandardSSDLRS),
+						string(compute.UltraSSDLRS),
+					}, false),
+					DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
Is this case insensitive due to an API casebug?

Reply:
Unfortunately yes, it's prevalent through all the resources that use this parameter.

Reply:
If we can open an issue on the SDK and link it here that would be great